src/basic/socket-util.c

```diff
@@ -905,6 +905,11 @@ ssize_t receive_one_fd_iov(
         if (k < 0)
                 return (ssize_t) -errno;
 
+        if (FLAGS_SET(mh.msg_flags, MSG_CTRUNC)) { /* Too many fds? */
+                cmsg_close_all(&mh);
+                return -ENOBUFS;
+        }
```

Comment:
upcoming version has the helper (though I called it recvmsg_safe()).

THe _cleanup_ thing would only be useful in one case, and isn't pretty since the control data is typically not initialized before recvmsg() returns successfully. This means we'd have to do an ugly ptr assignment when we consider it valid, for the cleanup stuff to get activated, but not sure it's worth it if it only helps a single place... so i left this out for now.